quantumnetworks/systems/doublemode.py

```diff
@@ -89,11 +93,54 @@ def default_B_in(self, t: float):
         omega_b = self.params["omega_b"]
         return np.exp(1j * (omega_b * t))
 
+    # Nonlinear
+    # =================================
+    def A_nl(self, x: np.ndarray):
```

Comment:
Minor nomenclature thing, but maybe we could call these functions `f_nl` and `Jf_nl` (instead of `A_nl` and `A_nl_jf`) since they are evaluating the function `f(x, p, u)`.